src/transformers/models/superglue/modeling_superglue.py

```diff
@@ -0,0 +1,845 @@
+    bins1 = alpha.expand(b, 1, n)
+    alpha = alpha.expand(b, 1, 1)
+
+    couplings = torch.cat([torch.cat([scores, bins0], -1), torch.cat([bins1, alpha], -1)], 1)
+
+    norm = -(ms + ns).log()
+    log_mu = torch.cat([norm.expand(m), ns.log()[None] + norm])
+    log_nu = torch.cat([norm.expand(n), ms.log()[None] + norm])
+    log_mu, log_nu = log_mu[None].expand(b, -1), log_nu[None].expand(b, -1)
+
+    Z = log_sinkhorn_iterations(couplings, log_mu, log_nu, iterations=iterations)
+    Z = Z - norm  # multiply probabilities by M+N
+    return Z
+
+
+def arange_like(x, dim: int):
+    return x.new_ones(x.shape[dim]).cumsum(0) - 1
+
+
+@dataclass
+class KeypointMatchingOutput(ModelOutput):
+    """
+    Base class for outputs of keypoint matching models. Due to the nature of keypoint detection and matching, the number of
+    keypoints is not fixed and can vary from image to image, which makes batching non-trivial. In the batch of images,
+    the maximum number of matches is set as the dimension of the matches and matching scores. The mask
+    tensor is used to indicate which values in the keypoints, matches and matching_scores tensors are keypoint matching
+    information.
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*):
+            Loss computed during training.
+        mask (`torch.BoolTensor` of shape `(batch_size, num_keypoints)`):
+            Mask indicating which values in matches and matching_scores are keypoint matching information.
+        matches (`torch.FloatTensor` of shape `(batch_size, 2, num_matches)`):
+            Index of keypoint matched in the other image.
+        matching_scores (`torch.FloatTensor` of shape `(batch_size, 2, num_matches)`):
+            Scores of predicted matches.
+        keypoints (`torch.FloatTensor` of shape `(batch_size, num_keypoints, 2)`):
+            Absolute (x, y) coordinates of predicted keypoints in a given image.
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    mask: torch.FloatTensor = None
+    matches: torch.FloatTensor = None
+    matching_scores: torch.FloatTensor = None
+    keypoints: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+class SuperGlueMultiLayerPerceptron(nn.Module):
+    def __init__(
+        self,
+        config: SuperGlueConfig,
+        channels: List[int],
+        do_batch_norm: bool = True,
+    ):
+        super().__init__()
+        num_layers = len(channels)
+        layers = []
+        for i in range(1, num_layers):
+            layers.append(nn.Conv1d(channels[i - 1], channels[i], kernel_size=1, bias=True))
+            if i < (num_layers - 1):
+                if do_batch_norm:
+                    layers.append(nn.BatchNorm1d(channels[i]))
+                layers.append(nn.ReLU())
+        self.layers = nn.Sequential(*layers)
+        nn.init.constant_(self.layers[-1].bias, 0.0)
```

Comment:
Same here re weight initialization

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/29886/commits/995fbdb39465d7f3ad4e25570a8cabfc618c6346)